text/0004-optional-frontend.md

```diff
@@ -0,0 +1,45 @@
+- Title: Optional Meilisearch Front-end
+- Start Date: 2020/11/16
+- specification PR:
+- Meilisearch Issue: #411
+
+# Optional Meilisearch Front-end
+
+## First section: Feature Description and Interaction
+
+### Summary
+
+For testing purposes, MeiliSearch is shipped with a frontend that is served on the server's root. This provides the user with a basic GUI to perform searches on his document collection. While this is useful in a development environment, this is not necessarily desirable in a production environment. This specification propose to make this frontend optional thanks to binary setting.
+
+### Motivation
+
+In most production cases, there is no need for this frontend, and exposing it is a bad practice. We want to provide an opt-out option for this feature.
+
+### Prior Art and R&D
+
+N.A
+
+### Explanation
+
+On MeiliSearch startup, we would like to pass a flag `--disable-web-gui` that disables the web user interface.
+
+### Impact on documentation
+
+This new feature adds a new flag to the binary, that needs to be documented [here](https://docs.meilisearch.com/guides/advanced_guides/configuration.html).
```

Comment:
It should also be explained:
- In the [environment documentation](https://docs.meilisearch.com/guides/advanced_guides/configuration.html#environment) 
- In the [web interface documentation](https://docs.meilisearch.com/guides/advanced_guides/web_interface.html#web-interface)